src/app/server/Server.h

```diff
@@ -276,44 +278,44 @@ struct CommonCaseDeviceServerInitParams : public ServerInitParams
         this->accessDelegate = Access::Examples::GetAccessControlDelegate();
 
         // Inject ACL storage. (Don't initialize it.)
-        this->aclStorage = &sAclStorage;
+        this->aclStorage = &sAclStorage.get();
 
 #if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
         ChipLogProgress(AppServer, "Initializing subscription resumption storage...");
-        ReturnErrorOnFailure(sSubscriptionResumptionStorage.Init(this->persistentStorageDelegate));
-        this->subscriptionResumptionStorage = &sSubscriptionResumptionStorage;
+        ReturnErrorOnFailure(sSubscriptionResumptionStorage->Init(this->persistentStorageDelegate));
+        this->subscriptionResumptionStorage = &sSubscriptionResumptionStorage.get();
 #else
         ChipLogProgress(AppServer, "Subscription persistence not supported");
 #endif
 
 #if CHIP_CONFIG_ENABLE_ICD_CIP
         if (this->icdCheckInBackOffStrategy == nullptr)
         {
-            this->icdCheckInBackOffStrategy = &sDefaultICDCheckInBackOffStrategy;
+            this->icdCheckInBackOffStrategy = &sDefaultICDCheckInBackOffStrategy.get();
         }
 #endif
 
         return CHIP_NO_ERROR;
     }
 
 private:
-    static KvsPersistentStorageDelegate sKvsPersistenStorageDelegate;
-    static PersistentStorageOperationalKeystore sPersistentStorageOperationalKeystore;
-    static Credentials::PersistentStorageOpCertStore sPersistentStorageOpCertStore;
-    static Credentials::GroupDataProviderImpl sGroupDataProvider;
-    static chip::app::DefaultTimerDelegate sTimerDelegate;
-    static app::reporting::ReportSchedulerImpl sReportScheduler;
+    static Global<KvsPersistentStorageDelegate> sKvsPersistenStorageDelegate;
```

Comment:
The `CommonCaseDeviceServerInitParams` is meant to be for in-tree examples only, if the aim was to reduce stack usage in a specific platform app, couldn't implementing a platform specific `ServerInitParams` achieve the same result without modifying the generic server? Or does the way the resources are spread out in the Server.cpp / Server.h prevent that?

Reply:
The aim is to reduce the memory used by global variables (not stack), especially ones that are not used. I'm not sure under what conditions the compiler/linker is allowed to optimise them away or what needs to be done to ensure that it does. At least some of those were certainly taking up space in our firmware and AFAICT we're not using CommonCaseDeviceServerInitParams explicitly. Our platform is esp32.

Reply:
If these objects were taking space even without using `CommonCaseDeviceServerInitParams` there is indeed something going wrong with the optimisation. Might be worth opening an issue in the future. otherwise LGTM.